src/Skia/Perspex.Skia/RenderTarget.cs

```diff
@@ -4,24 +4,187 @@
+//
+#if __IOS__
+using CoreGraphics;
+using UIKit;
+#elif WIN32
+using Perspex.Win32.Interop;
+#endif
 
 namespace Perspex.Skia
 {
-#if !__ANDROID__
-    class RenderTarget : PerspexHandleHolder, IRenderTarget
+    internal class RenderTarget : IRenderTarget
+    {
+        public SKSurface Surface { get; protected set; }
+
+        //protected override void Delete(IntPtr handle) => MethodTable.Instance.DisposeRenderTarget(handle);
+
+        public virtual DrawingContext CreateDrawingContext()
+        {
+            return
+                new DrawingContext(
+                    new DrawingContextImpl(Surface.Canvas));	// MethodTable.Instance.RenderTargetCreateRenderingContext(Handle)));
+        }
+
+        public void Dispose()
+        {
+            //throw new NotImplementedException();
+        }
+    }
+
+    internal class WindowRenderTarget : RenderTarget
     {
-        public RenderTarget(IPlatformHandle handle) : base(MethodTable.Instance.CreateWindowRenderTarget(handle.Handle))
+        private readonly IntPtr _hwnd;
+        SKBitmap _bitmap;
+
+        int Width { get; set; }
+        int Height { get; set; }
+
+        public WindowRenderTarget(IntPtr hwnd)
         {
+            _hwnd = hwnd;
+            FixSize();
         }
 
-        protected override void Delete(IntPtr handle) => MethodTable.Instance.DisposeRenderTarget(handle);
+#if __IOS__
+        private CGRect GetApplicationFrame()
+        {
+            // if we are excluding Status Bar then we use ApplicationFrame
+            // otherwise we use full screen bounds. Note that this must also match
+            // the Skia/PerspexView!!!
+            //
+            bool excludeStatusArea = false; // TODO: make this configurable later
+            if (excludeStatusArea)
+            {
+                return UIScreen.MainScreen.ApplicationFrame;
+            }
+            else
+            {
+                return UIScreen.MainScreen.Bounds;
+            }
+        }
+#endif
 
-        public DrawingContext CreateDrawingContext()
+        void FixSize()
         {
+            int width, height;
+            GetPlatformWindowSize(_hwnd, out width, out height);
+            if (Width == width && Height == height)
+                return;
+
+            Width = width;
+            Height = height;
+
+            if (Surface != null)
+            {
+                Surface.Dispose();
+            }
+
+            if (_bitmap != null)
+            {
+                _bitmap.Dispose();
+            }
+
+            _bitmap = new SKBitmap(width, height, SKColorType.N_32, SKAlphaType.Premul);
+
+            //bitmap.LockPixels();
+            IntPtr length;
+            var pixels = _bitmap.GetPixels(out length);
+
+            // Wrap the bitmap in a Surface and keep it cached
+            Surface = SKSurface.Create(_bitmap.Info, pixels, _bitmap.RowBytes);
+        }
+
+        void GetPlatformWindowSize(IntPtr hwnd, out int w, out int h)
+        {
+#if __IOS__
```

Comment:
I'm not sure I like the idea of having the platform-specific stuff like this included/excluded based on `#if`s. Perspex.Skia is a shared project, so it could call out to platform-specific classes in the platform-specific project that it's included in right?

This probably doesn't need to be done for the initial merge, but it's something to discuss.